racksdb/web/ui/src/views/DatacenterRoomView.vue

```diff
@@ -0,0 +1,106 @@
+<script setup lang="ts">
+import { useHttp } from '@/plugins/http';
+import { ref, onMounted, inject } from 'vue'
+import SearchBar from '@/components/SearchBar.vue';
+import { injectionKey } from '@/plugins/runtimeConfiguration';
+import type { Ref } from 'vue'
+import type { Datacenter } from './DatacentersView.vue';
+
+const datacenters: Ref<Array<Datacenter>> = ref([])
+const datacenterDetails: Ref<Datacenter | undefined> = ref()
+const racks: Ref<Array<Rack>> = ref([])
+const rackDetails: Ref<Array<Rack>> = ref([])
+const showFullImg = ref(false)
+const http = useHttp()
+
+function openImg(){
```

Comment:
This function is also used to close image modal. A better name would be `toggleImageModal()`.